code/Test_definitions/connected-network-type-subscriptions.feature

```diff
@@ -0,0 +1,276 @@
+    Then the response code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_PROTOCOL"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_15_invalid_credential
+  Scenario: subscription creation with invalid credential type
+    Given a valid subscription request body
+    And the request property "$.credentialType" is not "ACCESSTOKEN"
+    When the request "createSubscription" is sent
+    Then the response code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_CREDENTIAL"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_16_invalid_access_token_type
+  Scenario: subscription creation with invalid access token type
+    Given a valid subscription request body
+    And the request property "$.accessTokenType" is not "bearer"
+    When the request "createSubscription" is sent
+    Then the response code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_TOKEN" or "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_17_no_authorization_header_for_create_subscription
+  Scenario: subscription creation with invalid credentials
+    Given a valid subscription request body
+    And header "Authorization" token is set to invalid credentials
+    When the request "createSubscription" is sent
+    Then the response code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_18_expired_access_token_for_create_subscription
+  Scenario: Expired access token for create subscription
+    Given a valid subscription request body and header "Authorization" is expired
+    When the request "createSubscription" is sent
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+  @connected_network_type_subscriptions_19_invalid_inconsistent_access_token
+  Scenario: subscription creation with inconsistent access token for requested events subscription
+   # To test this, a token have to be obtained for a different device
+    Given a valid subscription request body
+    And the request body property "$.device" is set to a valid testing device supported by the service
+    And header "Authorization" set to access token referring different device
+    When the request "createSubscription" is sent
+    Then the response code is 403
+    And the response property "$.status" is 403
+    And the response property "$.code" is "SUBSCRIPTION_MISMATCH"
```

Comment:
This is not to me the UC for SUBSCRIPTION_MISMATCH. This error is designed to cover the scenario where an API Client requests the creation of an event subscription for an event type for which it does not have permission. The token is valid and the client can consume the endpoint, but only for the events that were requested in the authentication.